ledger/complete/ledger.go

```diff
@@ -183,7 +183,11 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, err error) {
 	return ledger.State(newRootHash), nil
 }
 
-// Prove provides proofs for a ledger query and errors (if any)
+// Prove provides proofs for a ledger query and errors (if any).
+//
+// Prove must provide proofs in an order not correlated to the register ID order in the query.
```

Comment:
Isn't "not correlated" a good description of what the API should achieve? I am referring to the statistical meaning of non-correlation : the order is different when we look at the samples overall, but it can be equal over a few samples.